R/PipeOpEncodePL.R

```diff
@@ -0,0 +1,151 @@
+#' @title Factor Encoding
+#'
+#' @usage NULL
+#' @name mlr_pipeops_encode
+#' @format [`R6Class`][R6::R6Class] object inheriting from [`PipeOpTaskPreprocSimple`]/[`PipeOpTaskPreproc`]/[`PipeOp`].
+#'
+#' @description
+#' Encodes columns of type `numeric` and `integer`.
+#'
+#'
+#'
+#' Use the [`PipeOpTaskPreproc`] `$affect_columns` functionality to only encode a subset of columns, or only encode columns of a certain type.
+#'
+#' @section Construction:
+#' ```
+#' PipeOpEncodePL$new(task_type, id = "encodepl", param_vals = list())
+#' ```
+#' * `task_type` :: `character(1)`\cr
+#'
+#' * `id` :: `character(1)`\cr
+#'   Identifier of resulting object, default `"encode"`.
+#' * `param_vals` :: named `list`\cr
+#'   List of hyperparameter settings, overwriting the hyperparameter settings that would otherwise be set during construction. Default `list()`.
+#'
+#' @section Input and Output Channels:
+#' Input and output channels are inherited from [`PipeOpTaskPreproc`].
+#'
+#' The output is the input [`Task`][mlr3::Task] with all affected `numeric` and `integer` columns
+#'
+#' @section State:
+#' The `$state` is a named `list` with the `$state` elements inherited from [`PipeOpTaskPreproc`], as well as:
+#' * ` ` :: named `list`\cr
+#'
+#' @section Parameters:
+#' The parameters are the parameters inherited from [`PipeOpTaskPreproc`], as well as:
+#' * `method`  :: `character(1)` \cr
+#'   Initialized to `""`. One of:
+#'
+#' @section Methods:
+#' Only methods inherited from [`PipeOpTaskPreprocSimple`]/[`PipeOpTaskPreproc`]/[`PipeOp`].
+#'
+#' @references
+#' `r format_bib("gorishniy_2022")`
+#'
+#' @family PipeOps
+#' @template seealso_pipeopslist
+#' @include PipeOpTaskPreproc.R
+#' @export
+#' @examples
+#' library("mlr3")
+#'
+PipeOpEncodePL = R6Class("PipeOpEncodePL",
+  inherit = PipeOpTaskPreprocSimple,
+  public = list(
+    initialize = function(task_type, id = "encodepl", param_vals = list()) {
+      # NOTE: Might use different name, change assert, and conditions
+      assert_choice(task_type, mlr_reflections$task_types$task)
+      if (task_type == "TaskRegr") {
+        private$.tree_learner = LearnerRegrRpart$new()
+      } else if (task_type == "TaskClassif") {
+        private$.tree_learner = LearnerClassifRpart$new()
+      } else {
+        stopf("Task type %s not supported", task_type)
+      }
```

Comment:
Don‘t need `task_type` if we use `method = "quantiles"`, meaning that we should only check this in that case and don‘t throw an error, etc. We could set task type as `"Task"` by default …
However, `mlr_reflections$task_types` does not contain this as an acceptable name, so change the assert.